### ISSUE TITLE
Visual: Partially window appears when swapping workspaces for duration of animation

### ISSUE THREAD

**Issue by BinaryInk** [dawsers/hyprscroller#67]
# Problem
Severity: Minor, visual

When swapping workspaces with an animation, any partially visible window on the workspace you're leaving in the direction of the animation will appear on the new workspace until the animation finishes. 

# Reproduction
1. Open two windows taking up 2/3 of the screen (so one is overlapping on the right)
2. Move to any workspace "to the right" of the current workspace and observe.

Happens whether the slide animation is horizontal or vertical; left, right, up, or down.

I suspect this may or may not be related to #65 ? 

(Apologies. I'm just realizing I haven't had the opportunity to test out #65. Mixture of not having the opportunity to build the latest Hyprland-git (though I think I've seen an Arch update come through for Hyprland since then) and I've been working on my laptop ~99% of the time as of late)

**Comment by dawsers** [comment]
This is the same as #58 and #65. It is fixed in Hyprland v0.45.0. You can build Hyprland from git or wait until Arch releases it. It is currently in extra-testing, so it will probably take a couple of days at most.

**Comment by BinaryInk** [comment]
It appears this issue wasn't resolved with Hyprland >= v0.45.0.

#58 and #65 remain resolved, however

**Comment by dawsers** [comment]
I had to submit a PR to Hyprland to fix the other bug. There were too many assumptions and it was not easy to find a workaround. When a window is moving it is in one workspace for some amount of time, and then in the other. If you want to see it all the time, you will have to see other windows that also "are" in that same workspace. There is a small timing issue with the animation in Hyprland. As mentioned in other issues, hyprscroller extends the workspace beyond the boundary of the monitor, so Hyprland thinks some of its windows belong to nearby workspaces. I could fix the other bug because focus was "all the time" in the same workspace, and windows were not moving, but in this case the one that moves will toggle different workspaces visible. I think I will let this one stay.

Because of the nature of hyprscroller, I have had to add lots of hacks, and patch Hyprland's functions. You just need to look at the new overview code where things are scaled to see the amount of dynamic hooks I had to add to Hyprland's functions. That is very fragile, as any change may break functionality.

Unfortunately, it is what it is, and some times things cannot be fixed in the plugin. Trying to fix them in Hyprland works if the change can be minimal, like https://github.com/hyprwm/Hyprland/pull/8359 or https://github.com/hyprwm/Hyprland/pull/8423, but I cannot expect to be able to merge a huge change that can affect the basic functionality of Hyprland, because this plugin's reach is very niche.

Sorry.
